vend.js

```diff
@@ -1449,12 +1483,77 @@ var fetchAllProducts = function(args, connectionInfo, processPagedResults) {
   return processPagesRecursively(defaultArgs, connectionInfo, fetchProducts, processPagedResults);
 };
 
-var fetchPaginationInfo = function(args, connectionInfo){
-  if ( !(args && argsAreValid(args)) ) {
+var fetchProducts2 = function (args, connectionInfo, retryCounter) {
+  if (!retryCounter) {
+    retryCounter = 0;
+  } else {
+    log.debug('retry # ' + retryCounter);
+  }
+
+  var path = '/api/2.0/products';
+  var vendUrl = 'https://' + connectionInfo.domainPrefix + '.vendhq.com' + path;
+  var authString = 'Bearer ' + connectionInfo.accessToken;
+  log.debug('GET ' + vendUrl);
+  log.silly('Authorization: ' + authString); // TODO: sensitive data ... do not log?
+
+  //var domainPrefix = this.domainPrefix;
+
+  var options = {
+    url: vendUrl,
+    headers: {
+      'Authorization': authString,
+      'Accept': 'application/json'
+    },
+    qs: {
+      /*jshint camelcase: false */
+      after: args.after.value,
+      before: args.before.value,
+      deleted: args.deleted.value,
+      page_size: args.pageSize.value
+    }
+  };
+  if (args.page.value) {
+    log.debug('Requesting product page ' + args.page.value);
+  }
+
+  return sendRequest(options, args, connectionInfo, fetchProducts2, retryCounter);
+};
+
+var fetchAllProducts2 = function(connectionInfo, processPagedResults) {
+  var args = argsForInput.products.fetch2();
+  // args.after.value = 'id';
+  args.page.value = 1;
+  args.pageSize.value = 40000;
+  // args.active.value = true;
+
+  // set a default function if none is provided
+  if(!processPagedResults) {
+    processPagedResults = function processPagedResults(pagedData, previousData) {
+      log.debug('fetchAllProducts - default processPagedResults()');
+      if(previousData && previousData.length>0) {
+        //log.verbose(JSON.stringify(pagedData.products,replacer,2));
+        if(pagedData.data && pagedData.data.length>0) {
+          log.debug('previousData: ', previousData.length);
+          pagedData.data = pagedData.data.concat(previousData);
+          log.debug('combined: ', pagedData.data.length);
+        }
+        else {
+          pagedData.data = previousData;
+        }
+      }
+      return Promise.resolve(pagedData.data);
+    };
+  }
+  return processPagesRecursively(args, connectionInfo, fetchProducts2, processPagedResults);
+};
+
+
+var fetchPaginationInfo = function (args, connectionInfo) {
```

Comment:
Also add `args` to `fetchAllProducts2` so that we don't suffer later when a client cannot handle all the entries at the same time and wants them aggregated using a custom page size like 5k or 10k